httpx/decorators.py

```diff
@@ -0,0 +1,13 @@
+import inspect
+
+
+def request_signature(func):  # type: ignore
+    def wrapped(self, url, **kwargs):  # type: ignore
+        new_signature = inspect.signature(self.request)
+        new_signature.parameters["method"].replace(
+            kind=inspect.Parameter.POSITIONAL_ONLY
+        )
+        func.__signature__ = new_signature
+        return func(self, url, **kwargs)
+
+    return wrapped
```

Comment:
I don't think this is doing what you'd like it to do: if we say `client = httpx.Client()` then `client.get` will refer to `wrapped`, not the original function. We should assign `wrapped.__signature__` instead of `func.__signature__` — but there's no way to get access to `request()` from outside `wrapped`.

We *could* get it to work using:

```python
def share_signature(source: typing.Callable):
    sig = inspect.signature(source)

    def wrap(func):
        @functools.wraps(func)
        def wrapped(*args, **kwargs):
            return func(*args, **kwargs)

        wrapped.__signature__ = sig
        return wrapped

    return wrap
```

and then

```python
def request(self, ...): ...

@share_signature(request)
def get(self, ...): ...
```

At least, it would allow `help()` to display information about the signature. But autocompletion in the terminal (IPython) and VSCode still don't work.